cpu/sam0_common/periph/adc.c

```diff
@@ -48,11 +68,11 @@ static inline void _done(void)
 static bool _adc_syncing(void)
 {
 #ifdef CPU_SAMD21
-    if (ADC->STATUS.reg & ADC_STATUS_SYNCBUSY) {
+    if (ADC_DEV->STATUS.reg & ADC_STATUS_SYNCBUSY) {
         return true;
     }
-#else /* CPU_SAML21 */
-    if (ADC->SYNCBUSY.reg) {
+#else
+    if (ADC_DEV->SYNCBUSY.reg) {
         return true;
     }
 #endif
```

Comment:
All the other peripherals have something like

```C
static inline void _wait_syncbusy(void)
{
#ifdef ADC_STATUS_SYNCBUSY
    while (ADC_DEV->STATUS.reg & ADC_STATUS_SYNCBUSY) {}
#else
    while (ADC_DEV->SYNCBUSY.reg) {}
#endif
}
```

might as well use that here while you're at it.

Reply:
implemented it. In this way the explicit CPU differentiation is also eliminated, thats nice.